examples_tests/50.MeshPacking/main.cpp

```diff
@@ -0,0 +1,190 @@
+        auto pipelineLayout = driver->createGPUPipelineLayout(&pcRange, &pcRange + 1);
+
+        gpuPipeline = driver->createGPURenderpassIndependentPipeline(
+            nullptr, core::smart_refctd_ptr(pipelineLayout),
+            shaders, shaders + 2u,
+            vtxInputParams,
+            asset::SBlendParams(), asset::SPrimitiveAssemblyParams(), SRasterizationParams());
+    }
+}
+
+int main()
+{
+    // create device with full flexibility over creation parameters
+    // you can add more parameters if desired, check irr::SIrrlichtCreationParameters
+    irr::SIrrlichtCreationParameters params;
+    params.Bits = 24; //may have to set to 32bit for some platforms
+    params.ZBufferBits = 24; //we'd like 32bit here
+    params.DriverType = video::EDT_OPENGL; //! Only Well functioning driver, software renderer left for sake of 2D image drawing
+    params.WindowSize = dimension2d<uint32_t>(1280, 720);
+    params.Fullscreen = false;
+    params.Vsync = true; //! If supported by target platform
+    params.Doublebuffer = true;
+    params.Stencilbuffer = false; //! This will not even be a choice soon
+    auto device = createDeviceEx(params);
+
+    if (!device)
+        return 1; // could not create selected driver.
+
+
+    //! disable mouse cursor, since camera will force it to the middle
+    //! and we don't want a jittery cursor in the middle distracting us
+    device->getCursorControl()->setVisible(false);
+
+    //! Since our cursor will be enslaved, there will be no way to close the window
+    //! So we listen for the "Q" key being pressed and exit the application
+    QToQuitEventReceiver receiver;
+    device->setEventReceiver(&receiver);
+
+    auto* driver = device->getVideoDriver();
+    auto* smgr = device->getSceneManager();
+    auto* am = device->getAssetManager();
+    auto* fs = am->getFileSystem();
+
+    //
+    auto* qnc = am->getMeshManipulator()->getQuantNormalCache();
+    //loading cache from file
+    qnc->loadNormalQuantCacheFromFile<asset::CQuantNormalCache::E_CACHE_TYPE::ECT_2_10_10_10>(fs, "../../tmp/normalCache101010.sse", true);
+
+    // register the zip
+    device->getFileSystem()->addFileArchive("../../media/sponza.zip");
+
+    asset::IAssetLoader::SAssetLoadParams lp;
+    auto meshes_bundle = am->getAsset("sponza.obj", lp);
+    assert(!meshes_bundle.isEmpty());
+    auto mesh = meshes_bundle.getContents().begin()[0];
+    auto mesh_raw = static_cast<asset::ICPUMesh*>(mesh.get());
+
+    //saving cache to file
+    qnc->saveCacheToFile(asset::CQuantNormalCache::E_CACHE_TYPE::ECT_2_10_10_10, fs, "../../tmp/normalCache101010.sse");
+
+    core::vector<ICPUMeshBuffer*> meshBuffers;
+    for (uint32_t i = 0u; i < mesh_raw->getMeshBufferCount(); i++)
+        meshBuffers.push_back(mesh_raw->getMeshBuffer(i));
+
+    DrawIndexedIndirectInput mdiCallParams;
+    SVertexInputParams vtxInputParams;
+    packMeshBuffers(driver, meshBuffers, vtxInputParams, mdiCallParams);

```

Comment:
you should not care about `SVertexInputParams` in packer V2

Reply:
I know, this function uses mesh packer v1